Compile error with MSVC 14.1

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:** Godot 3.1 with 395ecc2235f91654ef7c06c7e73aabb352bb17fb
<!-- Specify commit hash if non-official. -->


**OS/device including version:**
<!-- Specify GPU model and drivers if graphics-related. -->
Windows 10 with Visual Studio 2017

```
PS F:\develop\godot> scons p=windows -j 4
scons: Reading SConscript files ...
Configuring for Windows: target=debug, bits=default
 Found MSVC version 14.1, arch amd64, bits=64
```

**Issue description:**
<!-- What happened, and what was expected. -->

The compiler complains
```
main\default_controller_mappings.gen.cpp(4): error C2466: cannot allocate an array of constant size 0
```

The generated content of this file as below
```c++
/* THIS FILE IS GENERATED DO NOT EDIT */
#include "default_controller_mappings.h"
#include "typedefs.h"
const char* DefaultControllerMappings::mappings[] = {
};
```

--- a/main/SCsub
+++ b/main/SCsub
@@ -74,7 +74,7 @@ def make_default_controller_mappings(target, source, env):
     platform_mappings = OrderedDict()
     for src in source:
         src_path = src.srcnode().abspath
-        with open(src_path, "rb") as f:
+        with open(src_path, "r") as f:
             # read mapping file and skip header
             mapping_file_lines = f.readlines()[2:]
 

